### PR TITLE
Moving ContentObject, User, Package Services

### PR DESCRIPTION
**Pull request by ryuyu**
This PR pulls the ContentObjectService, UserService, PackageService, and their dependencies into the library.

Please give this a careful look as there were merge conflicts.
I will be working closer to dev in the future, so hopefully, this will be the last one like this.

This was rebased onto dev off of a branch wayyyyy down from dev, so ignore the excess commits as they are already in dev, and squashing made them reappear here.

### CODE REVIEW

File: src/NuGetGallery.Services/Constants/GitConstants.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NuGetGallery.Services.Constants
+{
+    public static class GitConstants
```
Comment (scottbommarito) [inline src/NuGetGallery.Services/Constants/GitConstants.cs]: This file is very specific. I don't see us making any more `GitConstants`, so it seems a little unnecessary to make a whole file for this string. Is there another generic constants file that we can put this into? `ServiceConstants`?

Comment (ryuyu) [reply src/NuGetGallery.Services/Constants/GitConstants.cs]: You are absolutely right this should go into ServicesConstants.
Bit of background: this was initially moved here as I was planning on splitting the constants up a little bit, but that started to get unwieldy, so I just started moving stuff into ServicesConstants, and forgot to move this little bit. Good catch.

File: src/NuGetGallery.Services/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
[the remark this answers is not in the document]
Comment (ryuyu) [reply src/NuGetGallery.Services/Properties/AssemblyInfo.cs]: Tried, but there doesn't appear to be anything there for me. Formatted document and there were no changes either.

File: src/NuGetGallery.Services/Configuration/ILoginDiscontinuationConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Services.Entities;
+using NuGetGallery.Authentication;
+
+namespace NuGetGallery
+{
+    public interface ILoginDiscontinuationConfiguration
```
Comment (agr) [inline src/NuGetGallery.Services/Configuration/ILoginDiscontinuationConfiguration.cs]: Documentation for class and/or methods?

File: src/NuGetGallery.Services/Properties/AssemblyInfo.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (agr) [reply src/NuGetGallery.Services/Properties/AssemblyInfo.cs]: Windows line break VS Unix one?

**Pull request merged**